Assets/Scripts/Player.cs

```diff
@@ -3,25 +3,92 @@
 using System.Collections.Generic;
 using UnityEngine;
 
+// aight so what we want? 
+    // if MC, move according to controls
+    // if !MC: 
+        // if outside MC following position, get there quicker than movementspeed
+        // if very near position, normal follow
+
 public class Player : MonoBehaviour
 {
-    public float movementSpeed = 1f;
-    public float rotationSpeed = 1f;
-    private CharacterController characterController;
-    
+    [SerializeField]
+    private bool IsMainPlayer = false;
+    [SerializeField]
+    private float MovementSpeed = 1f;
+    [SerializeField]
+    private float RotationSpeed = 1f;
+    [SerializeField]
+    private float FollowDistance = 5f;
+    [SerializeField]
+    private float CatchUpSpeed = 2f; // ? TODO
+
+    private Vector3 TargetPosition;
+    private Quaternion TargetRotation;
+    private Vector3 Direction;
+
+    private bool IsMovementDisabled;
+
+    private CharacterController CharacterController;
+    private Player MainPlayer;
+    private Timer Timer;
+
     // Start is called before the first frame update
     void Start()
     {
-        characterController = GetComponent<CharacterController>();
+        CharacterController = GetComponent<CharacterController>();
+        var foundPlayers = GameObject.FindObjectsOfType<Player>();
+        MainPlayer = Array.Find(foundPlayers, player => player.IsMainPlayer);
+        Timer = GetComponent<Timer>();
     }
 
     // Update is called once per frame
     void Update()
     {
-        UpdateMovement();
+        if (IsMovementDisabled)
+            return;
+
+        if (IsMainPlayer)
+            UpdateMainCharacterMovement();
+        else
+            UpdateSecondaryCharacterMovement();
+    }
+
+    private void UpdateSecondaryCharacterMovement()
+    {
+        if (!Timer.IsInProgress())
+        {
+            Timer.StartTimer(.01f);
+
+            if (MainPlayer == null)
+            {
+                var foundPlayers = GameObject.FindObjectsOfType<Player>();
+                Debug.Log(foundPlayers.Length);
+                MainPlayer = Array.Find(foundPlayers, player => player.IsMainPlayer);
```

Comment:
Since the main player is always Sield, shouldn't searching an array for the main player only be done in an initialization method (ex.: Start)?

Reply:
As mentioned above, this is to prepare for a generalized MC / swap of main player. It only ever happens if MC == null, which should rarely happen.